module/zfs/zvol.c

```diff
@@ -847,8 +850,10 @@ zvol_discard(void *arg)
 		size = end - start;
 	}
 
-	if (start >= end)
-		goto out;
+	if (start >= end) {
+		error = SET_ERROR(EIO);
```

Comment:
I'm adding an additional `SET_ERROR(EIO)` here: also i don't understand why we need to check if `start >= end`.